Tde documentation updates for ga

Andriciuc opened the pull request:
initial PR

Andriciuc committed:
Added and reorg files

Added replication.md for replication implication in pg_tde, moved architecture.md in architecture folder and renamed to index.md to improve SEO and further refine doc structure.

Small fixes to features.md

Andriciuc committed:
Split Home and created TDE Overview

Home now points to index.md and added a new TOC link called TDE Overview which contains what index had, tde encrypts, limitations and versions, this will help a lot to further build up TDE in proper sections/topics rather than having a one big file.

Also updated mkdocs for this.

Andriciuc committed:
Further refined and split into topics

Split tde.md into:
What is TDE
How does TDE work
Why do you need it
Moved into how_to:
uninstall.md
decrypt
external param
multi-t-setup
Updated yaml structure, Overview now includes more details about TDE from above topics

Andriciuc committed:
Renamed files and folders for better SEO

Renamed a lot of the files in index and how_to (now how-to) to have better SEO

Andriciuc committed:
Merge remote-tracking branch 'origin/pg_tde_docs_updates_for_GA' into TDE-documentation-updates-for-GA

Andriciuc committed:
updated with Anders branch

Modified root to secret token access for mount point

Andriciuc committed:
Small updates to yml and apt

AndersAstrand commented on contrib/pg_tde/documentation/docs/features.md:
```diff
@@ -1,8 +1,12 @@
 # Features
 
-We provide `pg_tde` in two versions for both PostgreSQL Community and [Percona Server for PostgreSQL](https://docs.percona.com/postgresql/17/). The difference between the versions is in the set of included features which in its turn depends on the Storage Manager API. While PostgreSQL Community uses the default Storage Manager API, Percona Server for PostgreSQL extends the Storage Manager API ena ...
+`pg_tde` is available in two variants (or "flavors"):
```
This is apparently no longer true.

Andriciuc replied:
Indeed, thank you for pointing that out, I will update this.

Andriciuc replied:
Removed for RC2

Andriciuc committed:
updated index folder to reflect chapter name, overview

Andriciuc committed:
updated overview doc to index again

small fix to paths

Andriciuc committed:
updated links to reflect proper structure

I also updated the ToC links

Andriciuc committed:
Setup.md and structure changes

Split setup.md into setup.md (for local keys) and a separate tde install for global key manager with multiple KMS's.

Fixed a bit the ToC as links were renewed, updated content with small improvements for better flow.

Added key config for keyring, kmip and vault.

Andriciuc committed:
A handful of fixes

created principal key file and updated the flow of the docs, TOC, fixes to texts and updated the way we present sql commands

Andriciuc committed:
removed PG Community mentions

removed PG community tde_heap_basic mentions and improved tde limitation and supported versions text, I kept the Release notes mentions though. Also added a note saying tde_heap_basic is deprecated

Andriciuc committed:
updated cli and kmip server code

Updates merged from : PG-1506: Add separate parameters for kmip client cert and key #306

Andriciuc committed:
small fixes

Andriciuc committed:
Update functions.md

removed info for PG-1546 Remove grant managment functions #299

Andriciuc committed:
updated toc

also, improved CLI flow, split into 3 subchapters

Andriciuc committed:
small fixes to buttons added

Andriciuc committed:
smaller updates

improved doc for test.md and get-help.md fixed heading title for key providers.

Andriciuc committed:
smaller updates

implemented note for keyrings

Andriciuc committed:
Update requirements.txt

jeltz commented on contrib/pg_tde/documentation/docs/advanced-topics/index.md:
```diff
@@ -0,0 +1,9 @@
+
+This section covers the internal components and tools that power `pg_tde`.
+
+Use it to understand how encryption is implemented, fine-tune configuration using GUC variables, or leverage advanced CLI tools and functions for diagnostics and customization:
```
"using GUC variables" feels quite redundant. I think we could just remove that.

Andriciuc replied:
Resolved, now reads:

Use it to understand how encryption is implemented, fine-tune a configuration, leverage advanced CLI tools and functions for diagnostics and customization

Thank you for the catch!

Andriciuc replied:
Fixed it to look cleaner, thanks!

Use it to understand how encryption is implemented, fine-tune a configuration, leverage advanced CLI tools and functions for diagnostics and customization:

jeltz commented on contrib/pg_tde/documentation/docs/apt.md:
````diff
@@ -1,52 +1,51 @@
 
     ```{.bash data-prompt="$"}
-    $ sudo wget https://repo.percona.com/apt/percona-release_latest.generic_all.deb
+        $ sudo wget https://repo.percona.com/apt/percona-release_latest.generic_all.deb
````
Any reason for this change? Just a matter of taste? I am fine with either so do not mind jsut wondering if it matters.

Andriciuc replied:
Yep, just a simple indentation "fix", I find it easier to read like that, heard it helps. :)

jeltz commented on contrib/pg_tde/documentation/docs/apt.md:
````diff
@@ -60,19 +59,17 @@ Check the [list of supported platforms](install.md#__tabbed_1_1).
 
         ```sql
-        DROP EXTENSION pg_tde CASCADE
+            DROP EXTENSION pg_tde CASCADE
````
Not your fault but I think we should always end SQL statements with semi-colon, at least when we expect people to copy paste them. That way they will work nicely with copy paste into e.g. `psql`.

Andriciuc replied:
Ohh nice catch, updated with semicolon.

jeltz commented on contrib/pg_tde/documentation/docs/functions.md:
````diff
@@ -106,16 +95,15 @@ where:
-Use these functions to add a KMIP provider: 
+Use these functions to add a KMIP provider:
 
 ```
````
This block should also get the type SQL.

Andriciuc replied:
Fixed! Will add indentation later for clearer markdown code visibility

jeltz commented on contrib/pg_tde/documentation/docs/functions.md:
````diff
@@ -152,14 +140,14 @@ This function is intended for development or quick testing, and stores the keys
 Add a local keyfile provider:
 
-```
+```sql
````
If other code is indenteted shouldn't this be too? Or maybe we should just not indent any of them.

Andriciuc replied:
Definitely will be! I will make sure all files are indented properly, just takes a back priority for the moment. 👍

jeltz commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/index.md:
```diff
@@ -0,0 +1,16 @@
+In production environments, storing encryption keys locally on the PostgreSQL server can introduce security risks. To enhance security, `pg_tde` supports integration with external Key Management Systems (KMS) through a Global Key Provider interface.
+
+This section describes how you can configure `pg_tde` to use the local and external key providers.
+To use an external KMS with `pg_tde`, follow these two steps:
```
I think we usually do not have linebreaks in paragraphs so the two lines above should likely just be one single line.

Andriciuc replied:
I will keep it as is for now, it separates the paragraph better.

Andriciuc committed:
fixes part 1/2 from feedback review

Fixes courtesy of @AndersAstrand part 1/2

Andriciuc committed:
updated features.md

to remove deprecated PG Community version

Andriciuc committed:
removed compliance/regulations

removed text regarding compliance and regulatory alignment

Andriciuc committed:
fix for PG-1587

updated default value to false, from true

nastena1606 commented on contrib/pg_tde/documentation/docs/apt.md:
```diff
@@ -1,52 +1,51 @@
+This page explains how to install `pg_tde` with [Percona Distribution for PostgreSQL :octicons-link-external-16:](https://docs.percona.com/postgresql/latest/index.html).
 
-Check the [list of supported platforms](install.md#__tabbed_1_1).
+Make sure you check the [list of supported platforms](install.md#__tabbed_1_1) before continuing.
```
IMHO "make sure to check" is wordy. What's the difference between this phrase and just "Check"?

Andriciuc replied:
True, well thought 🤔 modified!

nastena1606 commented on contrib/pg_tde/documentation/docs/apt.md:
````diff
@@ -1,52 +1,51 @@
 
     ```{.bash data-prompt="$"}
-    $ sudo dpkg -i percona-release_latest.generic_all.deb
+        $ sudo dpkg -i percona-release_latest.generic_all.deb
````
Do you really want this much free space to the left? See the preview how it looks like

nastena1606 replied:
This applies to all commands

Andriciuc replied:
i think the preview looks okay, its more of a personal decision as I feel it is easier to read with indentations

Later edit: It seems that best practice is to not have spaces, I will fix this in the whole doc!

nastena1606 commented on contrib/pg_tde/documentation/docs/command-line-tools/index.md:
```diff
@@ -0,0 +1,7 @@
+# TDE CLI Tools
```
TDE stands for Transparent Data Encryption, it doesn't provide any CLI. pg_tde however, does. Consider renaming the title

Andriciuc replied:
Thank you for the catch! Fixed.

nastena1606 commented on contrib/pg_tde/documentation/docs/features.md:
```diff
@@ -1,20 +1,22 @@
+  * Index data for encrypted tables
+  * TOAST tables
+  * Temporary tables created during database operations
+    !!! note
```
```suggestion

    !!! note
```

nastena1606 replied:
The admonition is not rendered

Andriciuc replied:
fixed! I was looking into the notes and why some of them appear and some do not, I think I got it now why that happens

nastena1606 commented on contrib/pg_tde/documentation/docs/features.md:
```diff
@@ -1,20 +1,22 @@
+  * TOAST tables
+  * Temporary tables created during database operations
+    !!! note
+        Metadata of those tables is not encrypted
```
```suggestion
        Metadata of those tables is not encrypted
        
```

Andriciuc replied:
Fixed!

nastena1606 commented on contrib/pg_tde/documentation/docs/functions.md:
````diff
@@ -86,14 +75,14 @@ The Vault provider connects to a HashiCorp Vault or an OpenBao server, and store
 Use the following functions to add the Vault provider:
 
-```
+```sql
````
I suggest checking the code in the dark theme. There were issues with sql commands display. Fr=or this reason I removed code highlighting

Andriciuc replied:
Interesting, l use dark theme but no issues right now, still will keep an eye out, thank you!

nastena1606 commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/keyring.md:
```diff
@@ -0,0 +1,26 @@
+
+This setup is intended for development and stores the keys unencrypted in the specified data file. See [how to use external reference to parameters](../how-to/external-parameters.md) to add an extra security layer to your setup.
+
+!!! note
```
Same. you must add a space to have admonition rendered properly. https://squidfunk.github.io/mkdocs-material/reference/admonitions/#usage

Andriciuc replied:
Fixed! And I will keep that reference handy, thank you.

nastena1606 commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/set-principal-key.md:
```diff
@@ -0,0 +1,37 @@
+
+You can configure a default principal key using a global key provider. This key will be used by all databases that do not have their own encryption keys configured.
+
+## Create a Default Principal Key
```
You have different style for titles: Here's the Headline style while the next section has Sentence style. It makes sense to align it

nastena1606 replied:
Throughout the whole documentation.

nastena1606 commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/set-principal-key.md:
````diff
@@ -0,0 +1,37 @@
+        'ensure_new_key'
+    );
+```
+!!! note
````
Wrong syntax
```suggestion
!!! note

```

Andriciuc replied:
Fixed!

nastena1606 commented on contrib/pg_tde/documentation/docs/global-key-provider-configuration/vault.md:
```diff
@@ -0,0 +1,46 @@
+
+You can configure `pg_tde` to use HashiCorp Vault as a global key provider for managing encryption keys securely.
+
+!!! note
```
```suggestion
!!! note

```

Andriciuc replied:
Fixed!

nastena1606 commented on contrib/pg_tde/documentation/docs/index/how-does-tde-work.md:
```diff
@@ -0,0 +1,26 @@
+* **Internal encryption keys** to encrypt user data. They are stored internally, near the data that they encrypt.
+* The **principal key** to encrypt database keys. It is kept separately from the database keys and is managed externally in the key management store.
+
+!!! note
```
```suggestion
!!! note

```

nastena1606 commented on contrib/pg_tde/documentation/docs/index.md:
```diff
@@ -1,51 +1,13 @@
+`pg_tde` is the open source, community driven and futureproof PostgreSQL extension that provides Transparent Data Encryption (TDE) to protect data at rest. `pg_tde` ensures that the data stored on disk is encrypted, and that no one can read it without the proper encryption keys, even if they gain access to the physical storage media.
 
-Learn more [what Transparent Data Encryption is](tde.md#how-does-it-work) and [why you need it](tde.md#why-do-you-need-tde).
+Learn more about [what Transparent Data Encryption is](index/index.md) and [why do you need it](index/how-tde-helps.md).
```
I think this sentence is redundant. You have added the same stuff as buttons.

Andriciuc replied:
Indeed it is, I will stick with one style (showing buttons) instead of text as it feels kind of better. Fixed!

nastena1606 commented on contrib/pg_tde/documentation/docs/index/supported-versions.md:
```diff
@@ -0,0 +1,21 @@
+- **Regular updates and security patches** backed by Percona’s expert support team.
+- **Professional support** and guidance for secure PostgreSQL deployments.
+
+!!! note
```
```suggestion
!!! note

```

Andriciuc replied:
Note looks ok.

nastena1606 commented on contrib/pg_tde/documentation/docs/index/tde-limitations.md:
```diff
@@ -0,0 +1,9 @@
+# Limitations of TDE
```
of pg_tde?

Andriciuc replied:
Fixed!

nastena1606 commented on contrib/pg_tde/documentation/docs/release-notes/rc2.md:
```diff
@@ -0,0 +1,119 @@
+# pg_tde Release Candidate ({{date.RC}})
```
This will render the date for RC. You must create a new item in variables "RC2" or smth and reference it here

Andriciuc replied:
Yep, had to leave it a bit while working on these bigger ones. Fixed though!

nastena1606 commented on contrib/pg_tde/documentation/docs/setup.md:
```diff
@@ -1,137 +1,69 @@
+You can configure the `shared_preload_libraries` parameter in two ways:
+
+* Edit `shared_preload_libraries` manually:
+Add the following line to the `shared_preload_libraries` file:
```
```suggestion

    Add the following line to the `shared_preload_libraries` file:
```

Andriciuc replied:
Fixed! Kept the * line

nastena1606 commented on contrib/pg_tde/documentation/docs/setup.md:
````diff
@@ -1,137 +1,69 @@
+    ```
+
+* Use the [ALTER SYSTEM :octicons-link-external-16:](https://www.postgresql.org/docs/current/sql-altersystem.html) command:
+Run the following command in ```psql``` as a **superuser**:
````
Correct indentation.

nastena1606 commented on contrib/pg_tde/documentation/docs/setup.md:
````diff
@@ -1,137 +1,69 @@
+    ```
+
+* Use the [ALTER SYSTEM :octicons-link-external-16:](https://www.postgresql.org/docs/current/sql-altersystem.html) command:
+Run the following command in ```psql``` as a **superuser**:
````
```suggestion
Run the following command in `psql` as a **superuser**:
```

Andriciuc replied:
'Fixed both indentations and extra ''

nastena1606 commented on contrib/pg_tde/documentation/docs/setup.md:
```diff
@@ -1,137 +1,69 @@
-        * `port` is the port to communicate with the KMIP server. Typically used port is 5696.
-        * `server-certificate` is the path to the certificate file for the KMIP server.
-        * `client key` is the path to the client key.
+## 3. Create the extension in the Database
```
Why Database? It's confusing

Andriciuc replied:
Corrected this as it is superfluous, thank you!

nastena1606 commented on contrib/pg_tde/documentation/docs/setup.md:
```diff
@@ -1,137 +1,69 @@
+## 3. Create the extension in the Database
 
-        <i warning>:material-information: Warning:</i> Note that pg_tde_add_global_key_provider_kmip currently accepts only a combined client key + client certificate for the last parameter of this function named as `client key`.
+To do this after restarting PostgreSQL, create the extension in your database by running the [CREATE EXTENSION :octicons-link-external-16:](https://www.postgresql.org/docs/current/sql-createextension.html) command as a **superuser** or **database owner**. Connect to `psql` and type:
```
IMHO this sentence is too complex. I'd split it in two

nastena1606 commented on contrib/pg_tde/documentation/docs/setup.md:
````diff
@@ -1,137 +1,69 @@
+```
 
-    where:
+!!! note
````
```suggestion
!!! note

```

nastena1606 commented on contrib/pg_tde/documentation/docs/setup.md:
```diff
@@ -1,137 +1,69 @@
+    You can use external key providers to manage encryption keys. The recommended approach is to use the Key Management Store (KMS). See the next step on how to configure the KMS.
 
-## Next steps
+## Next step
```
Again, here you use sentence style. On other pages you have "Next **S**tep". Consider using the same style everywhere

nastena1606 commented on contrib/pg_tde/documentation/docs/test.md:
```diff
@@ -1,56 +1,60 @@
+
+    This forces PostgreSQL to check every tuple for visibility and reset the hint bits.
 
 !!! hint
```
```suggestion
!!! hint

```

nastena1606 commented on contrib/pg_tde/documentation/mkdocs.yml:
```diff
@@ -154,36 +154,54 @@ extra:
     - release-notes/alpha1.md
     - release-notes/mvp.md
-  - uninstall.md
+  - get-help.md
```
We've agreed to move this topic up so that it'll help generate leads.

Andriciuc committed:
fixes for nastia 1/2 and for jz 2/2

Andriciuc committed:
all fixes accounted for

fixed a lot of other minor issues but this is the last commit before merge

Andriciuc committed:
Resolve merge conflicts

nastena1606 commented on contrib/pg_tde/documentation/docs/advanced-topics/index.md:
```diff
@@ -0,0 +1,7 @@
+# Technical Reference
```
Title is misaligned with the rest

nastena1606 commented on contrib/pg_tde/documentation/docs/command-line-tools/pg-tde-change-key-provider.md:
```diff
@@ -0,0 +1,37 @@
+
+For example, you restore from an old backup and the address of the key provider changed in the meantime. You can use this tool to correct the configuration, allowing the server to start up.
+
+!!! warning
```
Wrong syntax

nastena1606 commented on contrib/pg_tde/documentation/docs/features.md:
```diff
@@ -1,20 +1,24 @@
-| Key management via: <br> - HashiCorp Vault; <br> - KMIP server; <br> - Local keyfile | Key management via: <br> - HashiCorp Vault; <br> - Local keyfile |
-| Logical replication of encrypted tables | |
+* Table encryption, including:
+  * Data tables
```
will be rendered as one list, not as nested one

Andriciuc committed:
fixes to titles

fixed title style across docs

Andriciuc committed:
fixed notes in features

and fixed mkdocs serve utility

Andriciuc committed:
updated function parameters

Updated from default true to false as per Commit f636e82 comments

AndersAstrand reviewed (approved):
Please squash before merging though :).
